### PR TITLE
Implement PXB-3155 - Move keyring vault to component 

### PR DESCRIPTION
**Pull request by altmannmarcelo**
https://jira.percona.com/browse/PXB-3155

Cherry-picked relevant commits:
https://github.com/percona/percona-server/commit/96b944152c7b2737fa317c695b07b584bac52bdd
https://github.com/percona/percona-server/commit/92926e1533384d2a666e4a8d50f43540eff30451

Adjusted PXB-related code to understand the vault as a component and fixed test cases.

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
## Manual Tests
We still don't have a valid PS version with vault as component, so no jenkins:

``` 
VAULT_URL=https://127.0.0.1:8200 VAULT_CACERT=${HOME}/vault.crt VAULT_CA=${HOME}/vault.crt VAULT_TOKEN=hvs.Aa83Us0I985GUVjjqCXVYbku ./run.sh -r /dev/shm/ -d /work/ps/ins/8.0/ -f -s keyring
Using /dev/shm//var as test root
Detecting server version...
Running against Percona Server 8.1.0-1-debug (XtraDB 8.1.0-1)
Using 'xtrabackup' as xtrabackup binary
Autodetected number of cores: 32
Limiting to 16 to avoid excessive resource consumption
Using 16 parallel workers
Running suite(s): keyring

==============================================================================
TEST                                   WORKER    RESULT     TIME(s) or COMMENT
------------------------------------------------------------------------------
innodb_keyring_kmip_component            w9	[skipped]   Keyring kmip requires KMIP Server configured
innodb_keyring_plugin_to_component       w9	[skipped]   Disabled until PXB-2969 is fixed
innodb_keyring_kms_component             w10	[skipped]   Keyring kms requires KMS variables configured
keyring_encryption                       w15	[skipped]   Requires server version lower than 8.0.30
bug1590351                               w2	[passed]    23
innodb_keyring_proxy_plugin              w9	[passed]    27
pxb-1705                                 w16	[passed]    35
innodb_encryption_mix_plugins            w5	[passed]    37
basic_encrytion                          w1	[passed]    44
pxb-1793                                 w2	[passed]    23
innodb_encryption_export                 w4	[passed]    54
pxb-1903                                 w16	[passed]    27
bug1737525                               w3	[passed]    67
pxb-1954                                 w2	[passed]    27
pxb-1937                                 w1	[passed]    30
pxb-1879                                 w9	[passed]    59
pxb-1936                                 w5	[passed]    50
pxb-2429                                 w4	[passed]    34
innodb_redo_log_encrypt                  w14	[passed]    98
shared_tablespace_encryption             w1	[passed]    39
reencrypt                                w3	[passed]    48
rollback                                 w2	[passed]    52
system_tablespace_encryption             w9	[passed]    41
keyring_pxb_2275                         w15	[passed]    143
innodb_keyring_vault_auto_v2             w11	[passed]    144
innodb_keyring_vault_auto_v1             w10	[passed]    146
innodb_keyring_vault_v2                  w13	[passed]    146
innodb_keyring_vault_v1                  w12	[passed]    151
xb_log_archiving_encrypt                 w5	[passed]    74
innodb_keyring_file_plugin               w8	[passed]    211
redo_log_archive                         w16	[passed]    173
innodb_keyring_file_both                 w6	[passed]    241
innodb_keyring_file_component            w7	[passed]    274
==============================================================================
Spent 2518 of 293 seconds executing testcases

SUMMARY: 33 run, 29 successful, 4 skipped, 0 failed

Skipped tests:  innodb_keyring_kmip_component innodb_keyring_plugin_to_component innodb_keyring_kms_component keyring_encryption

See results/ for detailed output

```

File: storage/innobase/xtrabackup/src/keyring_components.cc [hunk 1]
```diff
@@ -37,6 +37,8 @@ const char *XTRABACKUP_KEYRING_KMIP_CONFIG = "component_keyring_kmip.cnf";
 
 const char *XTRABACKUP_KEYRING_KMS_CONFIG = "component_keyring_kms.cnf";
 
+const char *XTRABACKUP_KEYRING_VAULT_CONFIG = "component_keyring_vault.cnf";
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/keyring_components.cc:40]: a side track question: what about other components? Did PS implement them and we tested those?

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/keyring_components.cc:40]: only manual testing too. There is no PS 8.1 for the bootstrap script to download from.

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/keyring_components.cc:40]: manual testing = PXB MTR ran locally

Comment (satya-bodapati) [reply storage/innobase/xtrabackup/src/keyring_components.cc:40]: I mean, did we cherry pick commits from PS for kms and kmip? similar to this work

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/keyring_components.cc:40]: Yes, those are part of 8.0 and were part of the initial 8.1 as part of the base branch.

Comment (satya-bodapati) [reply storage/innobase/xtrabackup/src/keyring_components.cc:40]: Thanks

File: storage/innobase/xtrabackup/test/inc/keyring_vault.sh [hunk 2]
```diff
@@ -37,18 +27,31 @@ function keyring_vault_ping()
...
+{
+  "vault_url": "${VAULT_URL}",
+  "secret_mount_point": "${VAULT_MOUNT_POINT}",
+  "secret_mount_point_version": "${VAULT_CONFIG_VERSION}",
+  "token": "${VAULT_TOKEN}",
+  "vault_ca": "${VAULT_CA}"
+}
 EOF
+  cp "${MYSQLD_DATADIR}/${binary}.my" ${instance_local_manifest}
+  cp "${MYSQLD_DATADIR}/component_keyring_vault.cnf" ${keyring_component_cnf}
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/test/inc/keyring_vault.sh:54]: this looks like new change? this is for server to pickup vault cnf?

**Review by satya-bodapati — APPROVED** [review]
LGTM!

File: storage/innobase/xtrabackup/test/inc/keyring_vault.sh (hunk 2 above)
Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/test/inc/keyring_vault.sh:54]: Yes, this is a common set of vars on all keyring include - For example https://github.com/percona/percona-xtrabackup/blob/trunk/storage/innobase/xtrabackup/test/inc/keyring_kms.sh#L47-L48

**Pull request merged**